source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -0,0 +1,513 @@
+		scan_->replaceRawDataStore(new AMCDFDataStore(AMUserSettings::userDataFolder % scan_->filePath(), false));
+
+		// Add all the old scan axes.
+		foreach (AMAxisInfo axis, scanAxes)
+			scan_->rawData()->addScanAxis(axis);
+
+		connect(this, SIGNAL(started()), this, SLOT(flushCDFDataStoreToDisk()));
+		connect(this, SIGNAL(cancelled()), this, SLOT(flushCDFDataStoreToDisk()));
+		connect(this, SIGNAL(failed()), this, SLOT(flushCDFDataStoreToDisk()));
+		connect(this, SIGNAL(finished()), this, SLOT(flushCDFDataStoreToDisk()));
+	}
+
+	// Get all the detectors added to the scan.
+	for (int i = 0, size = continuousConfiguration_->detectorConfigurations().count(); i < size; i++){
+
+		AMDetector *oneDetector = AMBeamline::bl()->exposedDetectorByInfo(continuousConfiguration_->detectorConfigurations().at(i));
+
+		if(oneDetector && scan_->rawData()->addMeasurement(AMMeasurementInfo(*oneDetector)))
+			scan_->addRawDataSource(new AMRawDataSource(scan_->rawData(), scan_->rawData()->measurementCount()-1), oneDetector->isVisible(), oneDetector->hiddenFromUsers());
+	}
+
+	connect(scanAssembler_, SIGNAL(actionTreeGenerated(AMAction3*)), this, SLOT(onScanningActionsGenerated(AMAction3*)));
+
+	if (scanAssembler_->generateActionTree())
+		buildScanControllerImplementation();
+
+	else
+		setFailed();
+}
+
+void AMContinuousScanActionController::flushCDFDataStoreToDisk()
+{
+	AMCDFDataStore *dataStore = qobject_cast<AMCDFDataStore *>(scan_->rawData());
+	if(dataStore && !dataStore->flushToDisk())
+		AMErrorMon::report(AMErrorReport(this, AMErrorReport::Serious, 38, "Error saving the currently-running scan's raw data file to disk. Watch out... your data may not be saved! Please report this bug to your beamline's software developers."));
+}
+
+#include "source/DataHolder/AMDSScalarDataHolder.h"
+#include "source/beamline/CLS/CLSAdvancedScalerChannelDetector.h"
+bool AMContinuousScanActionController::event(QEvent *e)
+{
+	if (e->type() == (QEvent::Type)AMAgnosticDataAPIDefinitions::MessageEvent){
+
+		AMAgnosticDataAPIMessage message = ((AMAgnositicDataEvent*)e)->message_;
+
+		switch(message.messageType()){
+
+		case AMAgnosticDataAPIDefinitions::AxisStarted:{
+
+			// Double check this!
+//			scan_->rawData()->beginInsertRows(continuousConfiguration_->scanAxisAt(0)->numberOfPoints(), -1);
+
+			break;}
+
+		case AMAgnosticDataAPIDefinitions::AxisFinished:{
+			// An argument could be made to put the control axis value stuff here.
+			if(generalConfig_->axisControlInfos().count() == 1)
+				axisFinished1DHelper();
+			else if(generalConfig_->axisControlInfos().count() == 2)
+				axisFinished2DHelper();
+
+
+			break;
+		}
+
+		case AMAgnosticDataAPIDefinitions::AxisValueFinished:
+
+			// Not relevant for continuous scans.
+			break;
+
+		case AMAgnosticDataAPIDefinitions::DataAvailable:{
+			AMAgnosticDataAPIDataAvailableMessage *dataAvailableMessage = static_cast<AMAgnosticDataAPIDataAvailableMessage*>(&message);
```

Comment:
I believe we are in. It's not being dealt with right now and I think we need to set up some performance tests to figure out how best to deal with it.